crates/bevy_ui/src/geometry.rs

```diff
@@ -0,0 +1,36 @@
+use bevy_reflect::Reflect;
+
+/// A rect, as defined by its "side" locations
+#[derive(Copy, Clone, PartialEq, Debug, Reflect)]
+#[reflect(PartialEq)]
+pub struct UiRect<T: Reflect + PartialEq> {
```

Comment:
Maybe out of scope for this PR, but what about adding some methods like `center() -> Point`, `translate(Vec2) -> Self`, ...? [Egui](https://github.com/emilk/egui/blob/0.17.0/emath/src/rect.rs) offers some nice methods on a `Rect` that I found quite useful.

Reply:
Totally useful, but we should keep this PR as small as possible to ease merging.

Reply:
Definitely out of scope for this PR. I have follow up PR's where I'll add a `new()` function to the `UiRect` and also update it's documentation. After those are merged we could think about adding some convenient methods to the `UiRect`.

@pubrrr You could make an issue for this so that we don't forget!

Reply:
-> #4323

Reply:
It should be noted that `Rect`/`UiRect` is not a rectangle in the geometry sense, but a struct that can hold information about the sides of a rectangle, like the width of the stroke

Reply:
Agreed. The documentation will get updated once these PR's are merged. [Here](https://github.com/bevyengine/bevy/pull/3503/files#diff-b33608d8591bbaf2e08e2b467eb467379dfdeb4ffca4b2369bc316d169e842edR119-R239) is the current documentation that would be added. I'll include your note as well. Thanks for the reminder!